File: exemplos/java/ConsultaIE.java
/**
 * FiscalAPI — Exemplo de consulta de Inscricao Estadual em Java
 * https://fiscalapi.com.br
 *
 * Requer Java 11+ (HttpClient nativo)
 */

import java.net.URI;
import java.net.URLEncoder;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.charset.StandardCharsets;

public class ConsultaIE {

    private static final String API_KEY = System.getenv("FISCALAPI_KEY");
    private static final String BASE_URL = "https://api.fiscalapi.com.br";

    private final HttpClient client = HttpClient.newHttpClient();

    /**
     * Consulta IE por UF e CPF.
     */
    public String consultarIE(String uf, String cpf) throws Exception {
        String url = String.format("%s/api/consultar?uf=%s&cpf=%s",
                BASE_URL,
                URLEncoder.encode(uf, StandardCharsets.UTF_8),
                URLEncoder.encode(cpf, StandardCharsets.UTF_8));

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("X-API-Key", API_KEY)
                .GET()
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() != 200) {
            throw new RuntimeException("Erro HTTP " + response.statusCode() + ": " + response.body());
        }

        return response.body();
    }

    /**
     * Consulta IE em todos os estados.
     */
    public String consultarIETodos(String cpf) throws Exception {
        String url = String.format("%s/api/consultar-ie-todos?cpf=%s",
                BASE_URL,
                URLEncoder.encode(cpf, StandardCharsets.UTF_8));

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("X-API-Key", API_KEY)
                .GET()
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

        if (response.statusCode() != 200) {
            throw new RuntimeException("Erro HTTP " + response.statusCode() + ": " + response.body());
        }

        return response.body();
    }

    public static void main(String[] args) throws Exception {
        ConsultaIE api = new ConsultaIE();

        // Consulta IE em Mato Grosso
        String resultado = api.consultarIE("MT", "12345678900");
        System.out.println("Resultado MT: " + resultado);

        // Consulta em todos os estados
        String todos = api.consultarIETodos("12345678900");
        System.out.println("Todos os estados: " + todos);
    }
}

File: exemplos/csharp/ConsultaIE.cs
/**
 * FiscalAPI — Exemplo de consulta de Inscricao Estadual em C# / .NET
 * https://fiscalapi.com.br
 *
 * Requer .NET 6+
 */

using System.Text.Json;

var apiKey = Environment.GetEnvironmentVariable("FISCALAPI_KEY")
    ?? throw new Exception("FISCALAPI_KEY nao definida");
var baseUrl = "https://api.fiscalapi.com.br";

using var client = new HttpClient();
client.DefaultRequestHeaders.Add("X-API-Key", apiKey);

// Consulta IE em Mato Grosso por CPF
async Task<JsonDocument> ConsultarIE(string uf, string cpf)
{
    var url = $"{baseUrl}/api/consultar?uf={Uri.EscapeDataString(uf)}&cpf={Uri.EscapeDataString(cpf)}";
    var response = await client.GetAsync(url);
    response.EnsureSuccessStatusCode();

    var json = await response.Content.ReadAsStringAsync();
    return JsonDocument.Parse(json);
}

// Consulta IE em todos os estados
async Task<JsonDocument> ConsultarIETodos(string cpf)
{
    var url = $"{baseUrl}/api/consultar-ie-todos?cpf={Uri.EscapeDataString(cpf)}";
    var response = await client.GetAsync(url);
    response.EnsureSuccessStatusCode();

    var json = await response.Content.ReadAsStringAsync();
    return JsonDocument.Parse(json);
}

// Perfil completo do produtor
async Task<JsonDocument> PerfilProdutor(string? cpf = null, string? cnpj = null)
{
    var param = cpf != null ? $"cpf={Uri.EscapeDataString(cpf)}" : $"cnpj={Uri.EscapeDataString(cnpj!)}";
    var url = $"{baseUrl}/api/perfil-produtor?{param}";
    var response = await client.GetAsync(url);
    response.EnsureSuccessStatusCode();

    var json = await response.Content.ReadAsStringAsync();
    return JsonDocument.Parse(json);
}

// Exemplo de uso
var resultado = await ConsultarIE("MT", "12345678900");
var results = resultado.RootElement.GetProperty("results");
Console.WriteLine($"Encontradas {results.GetArrayLength()} inscricoes em MT");

foreach (var ie in results.EnumerateArray())
{
    Console.WriteLine($"  IE: {ie.GetProperty("inscricao_estadual")} — {ie.GetProperty("situacao_ie")}");
    Console.WriteLine($"  Razao: {ie.GetProperty("razao_social")}");
    Console.WriteLine($"  Municipio: {ie.GetProperty("municipio")}\n");
}
